### ISSUE TITLE
Please save before fixing

### ISSUE THREAD

**Issue by adamreisnz** [AtomLinter/linter-eslint#972]
## Issue Type
Bug

## Issue Description
I occasionally see the message "Please save before fixing", without any further explanation.
This happens usually when I do a replace in file in multiple files across a project (with one or more of those files open in Atom).

Why does this message appear? What can be done about it? This didn't use to appear before a few versions back.

## Bug Checklist
- Restart Atom
Yes, also happens after restart.

- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
Not applicable here.

- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Atom version: 1.20.0-dev-19b1810
linter-eslint version: 8.2.1
ESLint version: 4.3.0
Hours since last Atom restart: 0
Platform: darwin
Using global ESLint from: /usr/local/lib/node_modules/eslint
Current file's scopes: [
  "source.js",
  "punctuation.terminator.statement.js"
]
linter-eslint configuration: {
  "fixOnSave": true,
  "useGlobalEslint": true,
  "lintHtmlFiles": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": []
}
```

**Comment by Arcanemagus** [comment]
I'm guessing that the method `find-and-replace` uses to modify a file you don't currently have open is to:
1. Open an editor to the file
2. Make the desired change
3. Save the file
4. Close the editor

If you have enabled the Fix on Save option, we add an observer for _all_ editors that triggers when they are saved. Before sending a `fix` job to the worker it [checks](https://github.com/AtomLinter/linter-eslint/blob/v8.2.1/src/main.js#L204) whether the editor is still valid, and that it hasn't been modified. I'm guessing that between triggering the save and Atom getting around to feeding the event to us, the editor has been closed and thus is no longer valid.

@IanVS do you think we should just silently exit if the editor isn't valid instead of telling the user something went wrong? I'm not sure we could handle this in any other way.

@adamreisnz You can safely ignore the message, it's mainly just there to yell at people that try to fix unsaved files 😉.

**Comment by adamreisnz** [comment]
Thanks, I figured it'd be something like that :)

**Comment by IanVS** [comment]
>do you think we should just silently exit if the editor isn't valid

I'd be fine with that.  Not getting a success message should be enough to indicate that nothing happened.  If we wanted to be a bit safer, we could also send a log to the console, just in case someone is confused and checks it.

**Comment by IanVS** [comment]
@adamreisnz do you want to take a crack at a PR?

**Comment by adamreisnz** [comment]
@IanVS I would if I had the time, but unfortunately between my contracting work, personal projects and open source projects that I'm already involved in I don't have the time currently to dig into any other packages, sorry!

**Comment by Arcanemagus** [comment]
Released in v8.3.0, sorry about the delay!
